packages/flutter/lib/src/rendering/table.dart

```diff
@@ -385,76 +385,48 @@ class RenderTable extends RenderBox {
        assert(configuration != null),
        _textDirection = textDirection,
        _columns = columns ?? (children != null && children.isNotEmpty ? children.first.length : 0),
-       _rows = rows ?? 0,
+       _rows = rows ?? (children != null ? children.length : 0),
        _columnWidths = columnWidths ?? HashMap<int, TableColumnWidth>(),
        _defaultColumnWidth = defaultColumnWidth,
        _border = border,
        _textBaseline = textBaseline,
        _defaultVerticalAlignment = defaultVerticalAlignment,
        _configuration = configuration {
-    _children = <RenderBox?>[]..length = _columns * _rows;
     this.rowDecorations = rowDecorations; // must use setter to initialize box painters array
-    children?.forEach(addRow);
+    if (children != null) {
+      final List<RenderBox> allChildren = <RenderBox>[];
+      for (final List<RenderBox> rows in children)
```

Comment:
`rows` makes it sound like each entry in `children` is multiple rows, but if i understand correctly, they are each a single row, so `row` would be more appropriate